src/github.com/matrix-org/go-neb/services/github/github.go

```diff
@@ -77,14 +78,69 @@ func (s *Service) requireGithubClientFor(userID string) (cli *gogithub.Client, r
 	return
 }
 
+const numberGithubSearchSummaries = 3
+const cmdGithubSearchUsage = `!github create owner/repo "search query"`
+
+func (s *Service) cmdGithubSearch(roomID, userID string, args []string) (interface{}, error) {
+	cli := s.githubClientFor(userID, true)
+	if len(args) < 2 {
+		return &gomatrix.TextMessage{"m.notice", "Usage: " + cmdGithubSearchUsage}, nil
+	}
+
+	query := fmt.Sprintf("repo:%s %s", args[0], strings.Join(args[1:], " "))
+	searchResult, res, err := cli.Search.Issues(query, nil)
+
+	if err != nil {
+		log.WithField("err", err).Print("Failed to search")
+		if res == nil {
+			return nil, fmt.Errorf("Failed to search. Failed to connect to Github")
+		}
+		return nil, fmt.Errorf("Failed to search. HTTP %d", res.StatusCode)
+	}
+
+	if searchResult.Total == nil || *searchResult.Total == 0 {
+		return &gomatrix.TextMessage{"m.notice", "No results found for your search query!"}, nil
+	}
+
+	numResults := *searchResult.Total
+	var htmlBuffer bytes.Buffer
+	var plainBuffer bytes.Buffer
+
+	numberOfSummaries := numberGithubSearchSummaries
+	if numResults < numberGithubSearchSummaries {
+		numberOfSummaries = numResults
+	}
+
+	summarizedIssues := searchResult.Issues[0:numberOfSummaries]
+
+	htmlBuffer.WriteString(fmt.Sprintf("Found %d results, here are the most relevant:<br><ol>", numResults))
+	plainBuffer.WriteString(fmt.Sprintf("Found %d results, here are the most relevant:\n", numResults))
+	for i, issue := range summarizedIssues {
+		if issue.HTMLURL == nil || issue.User.Login == nil || issue.Title == nil {
+			continue
+		}
+		htmlBuffer.WriteString(fmt.Sprintf(`<li><a href="%s" rel="noopener">%s: %s</a></li>`, *issue.HTMLURL, *issue.User.Login, *issue.Title))
```

Comment:
This is vulnerable to XSS attacks if you put arbitrary HTML as the issue title. Please escape appropriately.

Reply:
done